Sources/MuxStatsGoogleIMAPlugin/MuxImaListener.m

```diff
@@ -64,58 +64,100 @@ - (void)setupAdViewData:(MUXSDKAdEvent *)event withAd:(IMAAd *)ad {
     event.viewData = viewData;
 }
 
-- (MUXSDKAdEvent *_Nullable) dispatchEvent:(IMAAdEvent *)event {
+- (nullable MUXSDKAdEvent *)dispatchEvent:(IMAAdEvent *)event {
+
+    MUXSDKAdData *adData = [[MUXSDKAdData alloc] init];
+    if (event.ad != nil) {
+        adData.adId = event.ad.adId;
+        adData.adCreativeId = event.ad.creativeID;
+
+        // TODO: use newer IMA API here. universalAdIdValue
+        // is deprecated, but used for time being for parity
+        // with web&android
+        adData.adUniversalId = event.ad.universalAdIdValue;
+    }
+
+    return [self dispatchEvent:event.type
+                    withAdData:adData
+                 withIMAAdData:event.adData];
+}
+
+- (nullable MUXSDKAdEvent *)dispatchEvent:(IMAAdEventType)eventType
+                               withAdData:(nullable MUXSDKAdData *)adData
+                            withIMAAdData:(nullable NSDictionary *)imaAdData {
     MUXSDKAdEvent *playbackEvent;
-    
-    NSDictionary *adData = event.adData;
-    
-    switch(event.type) {
-        case kIMAAdEvent_STARTED:
+
+    switch(eventType) {
+        case kIMAAdEvent_STARTED: {
             if (_sendAdplayOnStarted) {
-                playbackEvent = [MUXSDKAdPlayEvent new];
+                playbackEvent = [[MUXSDKAdPlayEvent alloc] init];
                 [_playerBinding dispatchAdEvent: playbackEvent];
             } else {
                 _sendAdplayOnStarted = YES;
             }
-            playbackEvent = [MUXSDKAdPlayingEvent new];
+            playbackEvent = [[MUXSDKAdPlayingEvent alloc] init];
```

Comment:
Calling `alloc` and `init` is equivalent to `new`. `new` doesn't support custom initializers (initializers other than `init`, typically these take an argument). Modern Objective-C recommends use of custom initializers or factory methods (`+ (instancetype)factoryMethodA`) for more explicitness. [See here.](https://developer.apple.com/library/archive/releasenotes/ObjectiveC/ModernizationObjC/AdoptingModernObjective-C/AdoptingModernObjective-C.html#//apple_ref/doc/uid/TP40014150-CH1-SW8).

Reply:
Everything here [is compiled with Automatic Reference Counting enabled](https://clang.llvm.org/docs/AutomaticReferenceCounting.html), which disallows explicit management of reference counts except in limited circumstances. Bridging calls to older APIs. Not used in this SDK.